joint_state_broadcaster/src/joint_state_broadcaster.cpp

```diff
@@ -342,37 +352,45 @@ controller_interface::return_type JointStateBroadcaster::update(
       state_interface.get_value());
   }
 
-  joint_state_msg_.header.stamp = time;
-  dynamic_joint_state_msg_.header.stamp = time;
-
-  // update joint state message and dynamic joint state message
-  for (size_t i = 0; i < joint_names_.size(); ++i)
+  if (realtime_joint_state_publisher_ && realtime_dynamic_joint_state_publisher_->trylock())
```

Comment:
It looks like you mixed the publishers?

```suggestion
  if (realtime_joint_state_publisher_ && realtime_joint_state_publisher_->trylock())
```

Reply:
nice catch, thanks! :D